hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/RestResources.java

```diff
@@ -563,6 +564,183 @@ public Response associateMetrics(@PathParam("environmentId") String environmentI
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response getConfiguration(@PathParam("environmentId") String environmentId,
+                                     @PathParam("feedId") String feedId,
+                                     @PathParam("resourcePath") String resourcePath,
+                                     @DefaultValue("configuration") @QueryParam("dataType") DataEntity.Role dataType) {
+
+        return getConfigurationHelper(environmentId, feedId, resourcePath, dataType);
+    }
+
+    @GET
+    @javax.ws.rs.Path("/{environmentId}/resources/{resourcePath:.+}/data")
+    @ApiOperation("Retrieves the configuration of a resource")
+    @ApiResponses({
+                          @ApiResponse(code = 204, message = "OK"),
+                          @ApiResponse(code = 404, message = "Tenant, environment or resource doesn't exist",
+                                       response = ApiError.class),
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response getConfiguration(@PathParam("environmentId") String environmentId,
+                                     @PathParam("resourcePath") String resourcePath,
+                                     @DefaultValue("configuration") @QueryParam("dataType") DataEntity.Role dataType) {
+
+        return getConfigurationHelper(environmentId, null, resourcePath, dataType);
+    }
+
+    private Response getConfigurationHelper(String environmentId, String feedId, String resourcePath,
+                                               DataEntity.Role dataType) {
+        String tenantId = getTenantId();
+        CanonicalPath resource = composeCanonicalPath(tenantId, environmentId, feedId, resourcePath);
+
+        DataEntity data = inventory.inspect(resource, Resources.Single.class).data().get(dataType).entity();
+
+        return Response.ok(data).build();
+    }
+
+    @PUT
+    @javax.ws.rs.Path("/{environmentId}/{feedId}/resources/{resourcePath:.+}/data")
+    @ApiOperation("Updates the configuration of a resource")
+    @ApiResponses({
+                          @ApiResponse(code = 204, message = "OK"),
+                          @ApiResponse(code = 404, message = "Tenant, environment, resource or feed doesn't exist",
+                                       response = ApiError.class),
+                          @ApiResponse(code = 500, message = "Server error", response = ApiError.class)
+                  })
+    public Response updateConfiguration(@PathParam("environmentId") String environmentId,
+                                        @PathParam("feedId") String feedId,
+                                        @PathParam("resourcePath") String resourcePath,
+                                        @ApiParam(required = true) DataEntity.Update configuration) {
```

Comment:
probably yes, I put the role into the `DataEntity.Update` but it doesn't make sense, using the `@queryparam` and `DataEntity.Update - role` is better